builtin/providers/cloudstack/resources.go

```diff
@@ -53,6 +56,9 @@ func retrieveUUID(cs *cloudstack.CloudStackClient, name, value string) (uuid str
 	case "network":
 		uuid, err = cs.Network.GetNetworkID(value)
 	case "zone":
+		if value == UnlimitedResourceID {
+			return value, nil
+		}
```

Comment:
It is definitely supported by other resources, but not all. `project` for example does support `-1`. Not sure what's the best way to handle it in that case.